plugins/user-authenticators/ldap/src/main/java/org/apache/cloudstack/ldap/LdapAuthenticator.java

```diff
@@ -229,23 +235,27 @@ private Pair<Boolean, ActionOnFailedAuthentication> authenticate(String username
             processLdapUser(password, domainId, user, rc, ldapUser, accountType);
         } catch (NoLdapUserMatchingQueryException e) {
             logger.debug(e.getMessage());
-            // no user in ldap ==>> disable user in cloudstack
-            disableUserInCloudStack(user);
+            if (e.getMessage().contains(LDAP_READ_TIMED_OUT_MESSAGE) && !rc.first()) {
+                rc.second(ActionOnFailedAuthentication.INCREMENT_INCORRECT_LOGIN_ATTEMPT_COUNT);
+            } else {
+                // no user in ldap ==>> disable user in cloudstack
+                disableUserInCloudStack(user);
+            }
```

Comment:
(Sorry I may not have pressed buttons correctly last time)

Can this be moved to a new method as it is being used twice?